src/js/utils/reducers.js

```diff
@@ -68,3 +68,25 @@ export const remove = (state, payload) => {
         documents: reject(state.documents, ({ id }) => includes(payload, id)),
     };
 };
+
+/**
+ * Update the job fields of documents in state with matching job ids.
+ *
+ * @param state {object} the current reducer state
+ * @param payload {object} the job data to update
+ * @returns {object} the updated reducer state
+ */
+export function updateJobs(state, payload) {
+    if (!state.documents) {
+        return state;
+    }
+
+    forEach(state.documents, document => {
+        if (document.job.id === payload.id) {
+            document.job = {
+                ...document.job,
+                ...payload,
+            };
+        }
+    });
+}
```

Comment:
I am guessing redux-toolkit is making this non-mutating. In that case do you even need to return `state.

I think maybe this function already existed and you didn't write it...

Reply:
Uhhh yeah it does, but my code is different from what I intended in another way. I wanted the API to be consistent with how other document updaters worked. Redux-toolkit hid my mistake, but I am planning to rework to always return state with next push